File: oop/typescript/rideShare/Passenger.ts
import { Driver } from "./Driver";
import { Person } from "./Person";


export class Passenger extends Person {
  private drivers: Map<string, number>;

  constructor(name: string) {
    super(name);
    this.drivers = new Map<string, number>();
  }

  startRide(driverName: string) {
    this.drivers.set(driverName, (this.drivers.get(driverName) ?? 0) + 1);
    super.markRideStarted();
  }
}

File: oop/python/derek/ride_share/passenger.py

from person import Person

class Passenger(Person):
  
  def __init__(self, name) -> None:
    super().__init__(name)
    self.drivers = {}

  def start_ride(self, driver_name):
    self.drivers[driver_name] = self.drivers.get(driver_name, 0) + 1
    super().start_ride()